app/assets/javascripts/components/teacher/teacher_settings_page.es6.jsx

```diff
@@ -0,0 +1,287 @@
+        <RemoveInstrumentModal
+          isVisible={removeModalIsVisible}
+          handleClose={() => this.closeRemoveModal()}
+          fetchInstruments={() => this.fetchInstruments()}
+          instrument={instrument}
+        />
+      );
+    }
+  }
+
+  renderAddModal() {
+    const { addModalIsVisible } = this.state;
+    const { instruments } = this.state.person;
+    const { person } = this.props;
+
+    if (addModalIsVisible) {
+      return (
+        <AddInstrumentModal
+          isVisible={addModalIsVisible}
+          handleClose={() => this.closeAddModal()}
+          fetchInstruments={() => this.fetchInstruments()}
+          instruments={instruments}
+          instrumentable={person}
+        />
+      )
+    }
+  }
+
+  fetchInstruments() {
+    const route = ApiConstants.teachers.instruments(this.props.id);
+    const resolve = (response) => {
+      let person = this.state.person;
+      person.instruments = response.instruments;
+      this.setState({ person });
+    }
+    const reject = (response) => console.log(response);
+    Requester.get(
+      route,
+      resolve,
+      reject,
+    );
+  }
+
+  renderInstrument(instrument) {
+    return (
+      <div className="instrument">
+        <div className="instrumentName">
+          {instrument.name}
+        </div>
+        <a className="link"
+          onClick={() => this.openRemoveModal()}>
+          [Remove]
+        </a>
+        {this.renderRemoveModal(instrument)}
+      </div>
+    );
+  }
+
+  renderInstruments() {
+    const { instruments } = this.state.person;
+    if (instruments) {
+      return instruments.map((instrument) => this.renderInstrument(instrument));
+    }
+  }
+
+  saveAvailability(inputDate) {
+    const { calendar } = this.refs.settingsAvailability.refs
+    var eventArray = $(calendar).fullCalendar('clientEvents');
+
+    var availabilityArray = []
+    for (var i = 0; i < eventArray.length; i++) {
+      availabilityArray = availabilityArray.concat(range_to_array(eventArray[i]['start'], eventArray[i]['end']));
+    }
+    this.setState({ availability: availabilityArray});
+
+    const route = ApiConstants.teachers.update(this.props.id);
+    const params = {availability: availabilityArray};
+    const success = (response) => {
+      console.log("success");
+    };
+    const fail = (response) => {
+      console.log("fail");
+    };
+
+    Requester.update(
+        route,
+        params,
+        success,
+        fail
+    );
+  }
+
+  handleIntegerChange(event) {
+    let name = $(event.target).attr("name");
+    if (name === "teacher_school_level") {
+      name = "school_level";
+    }
+    var value = $(event.target).val();
+    value = parseInt(value);
+    this.setState({ [name] : value });
+  }
+
+  handleDatetimeChange(moment, name) {
+    if (name == 'birthday') {
+      this.setState({ birthday: moment });
+    } else if (name == 'waiver_date') {
+      this.setState({ waiver_date: moment });
+    }
+  }
```

Comment:
if `handleIntegerChange` and `handleDatetimeChange` are the same as the ones in `student_settings_page`, can you try putting these two in `UserSettings` instead to be more DRY? Not a huge issue, if it messes things up then we can look further into this later.